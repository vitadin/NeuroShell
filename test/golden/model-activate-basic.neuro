%% Test basic model activation functionality
\set[_echo_command="true"]

%% Create multiple models
\model-new[catalog_id="O4M"] first-model
\model-new[catalog_id="O3"] second-model
\model-new[catalog_id="O4M"] third-model

%% Test activation by exact name match
\model-activate first-model

%% Verify activation variables are set correctly
\get[#active_model_name]
\get[#active_model_id]
\get[#active_model_provider]
\get[#active_model_base]
\get[_output]

%% Activate a different model
\model-activate second-model

%% Verify variables updated
\get[#active_model_name]
\get[#active_model_provider]
\get[_output]