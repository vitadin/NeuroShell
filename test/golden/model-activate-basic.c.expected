Setting _echo_command = true
%%> "\\model-new[catalog_id=\"O4MC\"] first-model"
Created model 'first-model' (ID: 00000001, Provider: openai, Base: o4-mini)
%%> "\\silent \\openai-client-new[client_type=OAC]"
%%> "\\silent \\llm-client-activate OAC"
%%> "\\silent \\model-activate[id=true] 00000001-0000-4000-8000-000000000001"
%%> "\\model-new[catalog_id=\"O3R\"] second-model"
Created model 'second-model' (ID: 00000002, Provider: openai, Base: o3)
%%> "\\silent \\openai-client-new[client_type=OAR]"
%%> "\\silent \\llm-client-activate OAR"
%%> "\\silent \\model-activate[id=true] 00000002-0000-4000-8000-000000000002"
%%> "\\model-new[catalog_id=\"O4MC\"] third-model"
Created model 'third-model' (ID: 00000003, Provider: openai, Base: o4-mini)
%%> "\\silent \\openai-client-new[client_type=OAC]"
%%> "\\silent \\llm-client-activate OAC"
%%> "\\silent \\model-activate[id=true] 00000003-0000-4000-8000-000000000003"
%%> "\\model-activate first-model"
Activated model 'first-model' (ID: 00000001, Provider: openai, Base: o4-mini)
%%> "\\try \\silent \\llm-client-activate OAC"
%%> "\\silent \\llm-client-activate OAC"
%%> "\\try \\silent \\openai-client-new[client_type=OAC]"
%%> "\\silent \\openai-client-new[client_type=OAC]"
%%> "\\get[#active_model_name]"
#active_model_name = first-model
%%> "\\get[#active_model_id]"
#active_model_id = 00000001-0000-4000-8000-000000000001
%%> "\\get[#active_model_provider]"
#active_model_provider = openai
%%> "\\get[#active_model_base]"
#active_model_base = o4-mini
%%> "\\get[_output]"
_output = OpenAI client ready: OAC:b10394de (type: OAC)
%%> "\\model-activate second-model"
Activated model 'second-model' (ID: 00000002, Provider: openai, Base: o3)
%%> "\\try \\silent \\llm-client-activate OAR"
%%> "\\silent \\llm-client-activate OAR"
%%> "\\try \\silent \\openai-client-new[client_type=OAR]"
%%> "\\silent \\openai-client-new[client_type=OAR]"
%%> "\\get[#active_model_name]"
#active_model_name = second-model
%%> "\\get[#active_model_provider]"
#active_model_provider = openai
%%> "\\get[_output]"
_output = OpenAI client ready: OAR:b10394de (type: OAR)